crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs

```diff
@@ -447,3 +470,43 @@ pub fn annotated_assignment_default_in_stub(
     }
     checker.diagnostics.push(diagnostic);
 }
+
+// PYI052
+pub fn assignment_default_in_async_stub(
+    checker: &mut Checker,
+    targets: &[Expr],
+    value: &Expr,
+    annotation: Option<&Expr>,
+) {
```

Comment:
Rather than add a new method here, can we instead add the enum-related logic within `assignment_default_in_stub`? We're now checking `is_valid_default_value_without_annotation` multiple times, because it also gets called within `assignment_default_in_stub`. And we're avoiding a variety of conditions that need to be checked.

Reply:
Yeah this makes sense, I refactored it to now include the checks inside assignment_default_in_stub